packages/gatsby/src/query/__tests__/data-tracking.js

```diff
@@ -734,6 +734,10 @@ describe(`query caching between builds`, () => {
     }, 999999)
 
     it(`changing node to be used by any query triggers running that query (with restart)`, async () => {
+      // FIXME: this requires incremental index updates
+      if (process.env.GATSBY_EXPERIMENTAL_LMDB_INDEXES) {
+        return
+      }
```

Comment:
I dont quite understand how indexes implementation impact which queries get rerun 🤔

Reply:
So this test runs this query 3 times:

```graphql
{
  test(slug: { eq: "foo2" }) {
    slug
    content
  }
}
```

On the first run, node `foo2` doesn't exist, so we add a dependency for the whole `Test` connection.

On the second run, node `foo2` does exist so we should only add a dependency for this specific "foo2" node. But since the index is not updated, this second run doesn't find this node and we add a dependency for the whole `Test` connection again.

So in this test case (3rd run), we don't expect `static-query-1` query to run but it does run.

Reply:
Re-enabled this test in https://github.com/gatsbyjs/gatsby/pull/32261/commits/66d71841b67348634ddd315178769c8e482f7986 (temporary dropping indexes at the beginning of the build)